pkg/lib/bundlev2/synthesis.go

```diff
@@ -0,0 +1,715 @@
+	if err != nil {
+		return err
+	}
+
+	apiServiceDefs, err := b.getAPIServiceDefinitions()
+	if err != nil {
+		return err
+	}
+
+	nativeAPIs, err := b.getNativeAPIs()
+	if err != nil {
+		return err
+	}
+
+	b.synthesisedCSV = &v1alpha1.ClusterServiceVersion{
+		TypeMeta: v1.TypeMeta{
+			Kind:       CSVKind,
+			APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
+		},
+		ObjectMeta: v1.ObjectMeta{
+			// TODO: Are there other fields can be set here?
+			Name:         csvBasics.Name,
+			GenerateName: csvBasics.GenerateName,
+		},
+		Spec: v1alpha1.ClusterServiceVersionSpec{
+			InstallStrategy:           *installStrategy,
+			CustomResourceDefinitions: *specCRDs,
+			APIServiceDefinitions:     *apiServiceDefs,
+			NativeAPIs:                *nativeAPIs,
+			Version:                   csvBasics.Version,
+			MinKubeVersion:            csvBasics.MinKubeVersion,
+			Maturity:                  csvBasics.Maturity,
+			DisplayName:               csvBasics.DisplayName,
+			Description:               csvBasics.Description,
+			Keywords:                  csvBasics.Keywords,
+			Maintainers:               csvBasics.Maintainers,
+			Provider:                  csvBasics.Provider,
+			Links:                     csvBasics.Links,
+			Icon:                      csvBasics.Icon,
+			InstallModes:              csvBasics.InstallModes,
+			Replaces:                  csvBasics.Replaces,
+			Labels:                    csvBasics.Labels,
+			Annotations:               csvBasics.Annotations,
+			Selector:                  csvBasics.Selector,
+		},
+		// TODO: nothing can be set in status, right?
+	}
+	return nil
+}
+
+// loadStructuredObjects gets supported structured object recursively from bundle directory and put them into the map.
+// Map of the same bundle should always produce the same map considering directory walk is lexical ordered and therefore
+// input order is deterministic.
+// All supported type of structures are presented in supportedResources, extend the support types by adding to the map.
+func (b *bundlev2) loadStructuredObjects() error {
+	return filepath.Walk(b.bundleDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Recursively walking through every file in the bundle directory.
+		if !info.IsDir() {
+			reader, err := os.Open(path)
+			if err != nil {
+				return err
+			}
+
+			//if f.Name() == DependenciesFileName {
+			//	err = decodeFile(filepath.Join(metadata, f.Name()), dependenciesFile)
+			//	if err != nil {
+			//		log.Info("found dependencies file searching for csv")
+			//	} else {
+			//		log.Info("unable to parse dependencies.yaml file")
+			//	}
+			//}
+
+			bufObj := unstructured.Unstructured{}
+			decoder := yaml.NewYAMLOrJSONDecoder(reader, 30)
+			if err = decoder.Decode(&bufObj); err != nil {
+				b.logger.Warnf("file %s is not decodable as YAML or JSON, %v", info.Name(), err)
+				return nil
+			}
+
+			if IsSupported(bufObj.GetKind()) {
+				switch bufObj.GetKind() {
```

Comment:
can't these just be `b.appendTostructuredObjectMap(bufObj.GetKind, bufObj)`?

Reply:
fixed